src/PointwiseFunctions/AnalyticSolutions/Errors.hpp

```diff
@@ -0,0 +1,56 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Variables.hpp"
+#include "DataStructures/VariablesTag.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Tags {
+/*!
+ * \brief Prefix indicating the difference between two values, such as betwen a
+ * numeric quantity and the corresponding quantity computed from an analytic
+ * solution.
+ */
+template <typename Tag>
+struct Difference : db::PrefixTag, db::SimpleTag {
+  using type = db::const_item_type<Tag>;
+  using tag = Tag;
+};
+
+/*!
+ * \brief For each `Tag` in `TagsList`, where `TagsList` is a subset of
+ * `AnalyticTagsList`, compute the difference `Tag` - `Tags::Analytic<Tag>`.
+ */
+template <typename TagsList, typename AnalyticTagsList>
+struct DifferenceFromAnalyticCompute
+    : db::add_tag_prefix<::Tags::Difference, ::Tags::Variables<TagsList>>,
+      db::ComputeTag {
+  using argument_tags = tmpl::list<
+      Tags::Variables<TagsList>,
+      db::add_tag_prefix<Tags::Analytic, Tags::Variables<AnalyticTagsList>>>;
+  using return_type = ::Variables<db::wrap_tags_in<Tags::Difference, TagsList>>;
+
+  static constexpr void function(
+      const gsl::not_null<return_type*> error,
+      const ::Variables<TagsList>& numeric,
+      const ::Variables<db::wrap_tags_in<Tags::Analytic, AnalyticTagsList>>&
+          analytic) noexcept {
+    (*error) = return_type{numeric.number_of_grid_points()};
+    tmpl::for_each<TagsList>([&error, &numeric, &analytic](auto tag_v) {
```

Comment:
This loop should go over `AnalyticTagsList`

Reply:
AnalyticTagsList might be a superset of TagsList, so I don't understand how you can loop over AnalyticTagsList here

Reply:
Now I'm confused. I thought we have a set of system variables in, say, `VariablesTag`. Let `TagsList = db::get_variables_tags_list<VariablesTag>`. Now for a subset of these we have an analytic solution, this subset is `AnalyticTagsList`. In the DataBox we have `db::wrap_tags_in<::Tags::Analytic, AnalyticTagsList>` (may or may not be stored in a Variables). Are we on the same page here?

Reply:
Let's take the actual example in the test...maybe that will help?

TagsList are the tags we want to return, wrapped in Error<>; these are the quantities we want to compare to the analytic solution. I don't understand why we would only want to compare a subset of these to the analytic solution.

AnalyticCompute puts the analytic solution in the DataBox as a Variables; this is how the analytic solution will be added during a real simulation. But the analytic solutions in GR return many tags besides SpacetimeMetric, Phi, and Pi (namely a bunch of 3+1 quantities), and AnalyticCompute just adds all of these tags. So AnalyticTagsList can be a superset of TagsList.

Does this help? Maybe I'm confused somewhere. I'm happy to discuss further here or in a hangout